nuxeo-core/nuxeo-core-io/src/main/java/org/nuxeo/ecm/core/io/download/DownloadServiceImpl.java

```diff
@@ -219,6 +227,109 @@ public String getDownloadUrl(String storeKey) {
         return DownloadService.NXBIGBLOB + "/" + storeKey;
     }
 
+    /**
+     * Gets the base URL of a URL to use to download blobs
+     *
+     * @param urlString the URL to use to download blobs
+     *            ("http://localhost:8080/nuxeo/nxfile/default/3727ef6b-cf8c-4f27-ab2c-79de0171a2c8/files:files/0/file/image.png")
+     * @return the base URL with the context ("http://localhost:8080/nuxeo")
+     * @since 9.1
+     */
+    protected String getBaseDownloadUrl(String urlString) {
+        try {
+            URL url = new URL(urlString);
+            return url.getProtocol() + "://" + url.getHost() + (url.getPort() == -1 ? "" : ":" + url.getPort())
+                    + url.getFile().substring(0, url.getFile().indexOf('/', 1));
+        } catch (MalformedURLException e) {
+            return null;
+        }
+    }
+
+    @Override
+    public Pair<String, String> getDownloadPathAndActionFromUri(String uri) {
```

Comment:
As this becomes a public API still using a `Pair` is getting dirty. Maybe use a small static inner class `PathAndAction` with a `path` and an `action` field? The `action` could be an enum too.

Also it's not really a URI so let's not mention that in the method name -> `getDownloadPathAndAction` and `uri` -> `path` maybe?